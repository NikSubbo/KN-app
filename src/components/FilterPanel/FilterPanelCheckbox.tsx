import { FC } from 'react';

type Props = {
  label: string;
  isChecked: boolean;
  handleCheckListChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
};

export const FilterPanelCheckbox: FC<Props> = ({
  label,
  isChecked,
  handleCheckListChange
}) => {
  return (
    <div className="mr-6">
      <label
        htmlFor={label}
        className="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
      >
        <input
          type="checkbox"
          id={label}
          name={label}
          value={label}
          className="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          checked={isChecked}
          onChange={handleCheckListChange}
        />
        {label}
      </label>
    </div>
  );
};

export default FilterPanelCheckbox;
