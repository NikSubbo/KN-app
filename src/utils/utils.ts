export interface Product {
  productName: string;
  tags: string[];
  category: string;
  manufacturerUrl: string;
  description: string[];
  option1: string | null;
  option2: string | null;
}

export enum Categories {
  SoftwareDevelopment = 'Software Development',
  DailyBussiness = 'Daily Business',
  GraphicEditors = 'Graphic Editors',
  TextEditors = 'Text Editors',
  ManagementTools = 'Management Tools'
}

export const initialCheckboxes = {
  [Categories.SoftwareDevelopment]: false,
  [Categories.DailyBussiness]: false,
  [Categories.GraphicEditors]: false,
  [Categories.TextEditors]: false,
  [Categories.ManagementTools]: false
};

export const openNewUrl = (url: string) => {
  if (window) {
    window.open(url, '_blank');
  }
};

export const filterProducts = (
  products: Product[],
  search: string,
  checkboxes: Record<string, boolean>
) => {
  let updatedProducts = products;
  if (search) {
    updatedProducts = updatedProducts.filter((el: Product) =>
      el.productName.toLowerCase().includes(search.toLowerCase())
    );
  }
  if (!Object.values(checkboxes).every((el) => el === false)) {
    updatedProducts = updatedProducts.filter(
      (el: Product) => checkboxes[el.category]
    );
  }

  return updatedProducts;
};
