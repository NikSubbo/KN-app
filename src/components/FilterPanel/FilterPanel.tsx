import { FC } from 'react';
import FilterPanelCheckbox from './FilterPanelCheckbox';

export type Props = {
  handleCheckListChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  checkboxes: Record<string, boolean>;
};

export const FilterPanel: FC<Props> = ({
  handleCheckListChange,
  checkboxes
}) => {
  return (
    <div className="flex mt-8 mb-5 justify-between flex-wrap">
      {Object.entries(checkboxes).map(([label, isChecked]) => {
        return (
          <FilterPanelCheckbox
            key={label}
            label={label}
            isChecked={isChecked}
            handleCheckListChange={handleCheckListChange}
          />
        );
      })}
    </div>
  );
};

export default FilterPanel;
