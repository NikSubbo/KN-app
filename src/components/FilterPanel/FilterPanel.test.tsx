import { render, screen } from '@testing-library/react';
import { Categories, initialCheckboxes } from '../../utils/utils';
import FilterPanel, { Props } from './FilterPanel';

const DEFAULT_PROPS = {
  handleCheckListChange: jest.fn(),
  checkboxes: initialCheckboxes
};

const renderContainer = (overrideProps?: Partial<Props>) => {
  return render(<FilterPanel {...DEFAULT_PROPS} {...overrideProps} />);
};

describe('FilterPanel', () => {
  it('should match the snapshot', () => {
    const { container } = renderContainer();
    expect(container).toMatchSnapshot();
  });

  it('should check checkbox', () => {
    renderContainer({
      checkboxes: {
        ...initialCheckboxes,
        [Categories.SoftwareDevelopment]: true
      }
    });
    const softwareDevelopmentCheckbox = screen.getByLabelText(
      Categories.SoftwareDevelopment
    );
    expect(softwareDevelopmentCheckbox).toBeChecked();
  });
});
