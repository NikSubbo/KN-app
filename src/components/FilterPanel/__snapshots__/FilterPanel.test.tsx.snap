// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel should match the snapshot 1`] = `
<div>
  <div
    class="flex mt-8 mb-5 justify-between flex-wrap"
  >
    <div
      class="mr-6"
    >
      <label
        class="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
        for="Software Development"
      >
        <input
          class="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          id="Software Development"
          name="Software Development"
          type="checkbox"
          value="Software Development"
        />
        Software Development
      </label>
    </div>
    <div
      class="mr-6"
    >
      <label
        class="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
        for="Daily Business"
      >
        <input
          class="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          id="Daily Business"
          name="Daily Business"
          type="checkbox"
          value="Daily Business"
        />
        Daily Business
      </label>
    </div>
    <div
      class="mr-6"
    >
      <label
        class="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
        for="Graphic Editors"
      >
        <input
          class="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          id="Graphic Editors"
          name="Graphic Editors"
          type="checkbox"
          value="Graphic Editors"
        />
        Graphic Editors
      </label>
    </div>
    <div
      class="mr-6"
    >
      <label
        class="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
        for="Text Editors"
      >
        <input
          class="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          id="Text Editors"
          name="Text Editors"
          type="checkbox"
          value="Text Editors"
        />
        Text Editors
      </label>
    </div>
    <div
      class="mr-6"
    >
      <label
        class="inline-flex items-center text-darkGray font-medium text-sm cursor-pointer"
        for="Management Tools"
      >
        <input
          class="w-4 h-4 mr-2.5 accent-aqua cursor-pointer"
          id="Management Tools"
          name="Management Tools"
          type="checkbox"
          value="Management Tools"
        />
        Management Tools
      </label>
    </div>
  </div>
</div>
`;
